components/sync_manager/src/manager.rs

```diff
@@ -109,9 +108,7 @@ impl SyncManager {
     ) -> Result<SyncResult> {
         let key_bundle = sync15::KeyBundle::from_ksync_base64(&params.auth_info.sync_key)?;
         let tokenserver_url = url::Url::parse(&params.auth_info.tokenserver_url)?;
-        // TODO(issue 1684) this isn't ideal, we should have real support for interruption.
-        let p = Arc::new(AtomicUsize::new(0));
-        let interruptee = sql_support::SqlInterruptScope::new(p);
+        let interruptee = interrupt_support::ShutdownInterruptee;
```

Comment:
This should handle #1684.  Now we will interrupt syncing when when we enter shutdown mode.